File: Problems/1464. Maximum Product of Two Elements in an Array/1464. Maximum Product of Two Elements in an Array.java
class Solution {
    public int maxProduct(int[] nums) {
        int maxVal = -1;
        int sMaxVal = -1;
        for (int num: nums) {
            if (num >= maxVal) {
                sMaxVal = maxVal;
                maxVal = num;
            }
            else if (num > sMaxVal){
                sMaxVal = num;
            }
        }

        return (sMaxVal - 1) * (maxVal - 1);
    }
}

File: Problems/1464. Maximum Product of Two Elements in an Array/1464. Maximum Product of Two Elements in an Array.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        maxVal = -1;
        sMaxVal = -1;
        for num in nums:
            if (maxVal < sMaxVal): maxVal, sMaxVal = sMaxVal, maxVal;
            sMaxVal = max(num, sMaxVal);
            
        return (maxVal - 1) * (sMaxVal - 1);